aopy/preproc/base.py

```diff
@@ -729,3 +729,63 @@ def calc_eye_calibration(cursor_data, cursor_samplerate, eye_data, eye_samplerat
         return coeff, correlation_coeff, cursor_data_aligned, eye_data_aligned
     else:
         return coeff, correlation_coeff
+
+def calc_eye_target_calibration(eye_data, eye_samplerate, event_times, event_codes, target_pos,
+    align_events=range(81,89), penalty_events=[64], cursor_enter_center_target=80, offset=0., duration=0.1, return_datapoints=False):
+    """
+    Extracts eye data and calibrates, aligning eye data and calculating the least square fitting coefficients 
+    between eye positions and target positions
+    
+    Args:
+        eye_data (nt, 2 or 4): eye data in time (optionally for both left and right eyes)
+        eye_samplerate (float): sampling rate of the eye data
+        event_times (nevent): times at which events occur
+        event_codes (nevent): codes for each event
+        target_pos (ntarget, 3): N target positions (x,y,z)
+        align_events (list, optional): list of event codes to use for alignment. By default, align to
+            when the cursor enters 8 peripheral targets
+        penalty_events (list, optional): list of penalty event codes to exclude. By default,
+            events followed by hold penalties are excluded
+        cursor_enter_center_target (int, optional): event code for cursor_enter_center_target
+        offset (float, optional): time (in seconds) to offset from the given events. Positive offset yields later eye data
+        duration (float, optional): duration (in seconds) to average eye trajectories in the interval from the given events + offset
+        return_datapoints (bool, optional): if true, also returns cusor_data_aligned, eye_data_aligned
+
+    Returns:
+        tuple: tuple containing:
+            | **coefficients (neyech, 2):** coefficients [slope, intercept] for each eye channel
+            | **correlation_coeff (neyech):** correlation coefficients for each eye channel
+    """
+
+    # Get the corresponding cursor and eye data
+    trial_events, trial_times= trial_separate(event_codes, event_times, align_events, n_events=2)
+    if len(trial_times) == 0:
+        raise ValueError("Not enough trials to calibrate")
+    exclude_trials = np.array([(t[-1] in penalty_events) for t in trial_events], dtype='bool')
```

Comment:
This might be more flexible if we use “np.any(np.isin(t, penalty_events))” instead of “t[-1] in penalty_events”